src/EnergyPlus/Autosizing/CoolingWaterDesAirInletHumRatSizing.cc

```diff
@@ -0,0 +1,126 @@
+//     used to endorse or promote products derived from this software without specific prior
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <EnergyPlus/Autosizing/CoolingWaterDesAirInletHumRatSizing.hh>
+#include <EnergyPlus/DataAirSystems.hh>
+#include <EnergyPlus/DataEnvironment.hh>
+#include <EnergyPlus/DataSizing.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+#include <EnergyPlus/api/TypeDefs.h>
+
+#include <EnergyPlus/ReportSizingManager.hh>
+
+namespace EnergyPlus {
+
+void CoolingWaterDesAirInletHumRatSizer::initializeWithinEP(EnergyPlusData &state,
+                                                            std::string const &_compType,
+                                                            std::string const &_compName,
+                                                            bool const printWarningFlag)
+{
+    BaseSizer::initializeWithinEP(state, _compType, _compName, printWarningFlag);
+    this->sizingString = "Design Inlet Air Humidity Ratio";
+    this->dataDesInletAirHumRat = DataSizing::DataDesInletAirHumRat;
+    this->dataFlowUsedForSizing = DataSizing::DataFlowUsedForSizing;
+}
+
+EnergyPlus::AutoSizingResultType CoolingWaterDesAirInletHumRatSizer::size(EnergyPlusData &state, Real64 _originalValue)
+{
+    if (this->isNotInitialized) return AutoSizingResultType::ErrorType2;
```

Comment:
If a user calls initialize and then the sizer they are OK. If they only call the sizer ErrorType2 is returned. However, if they call Init, then the sizer, then update some of the Global data and call sizer again without Init first, using the same instance, they won't get the right answer. So, should I reset isNotInitialized here before the return to avoid that corner case?

Reply:
**Very** clever idea!  Yes!

Reply:
I'll add a reset for isNotInitialized here for all 3 sizers before moving on.